tls/s2n_early_data.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "tls/s2n_early_data.h"
+
+#include "tls/s2n_psk.h"
+#include "utils/s2n_safety.h"
+#include "utils/s2n_mem.h"
+
+S2N_CLEANUP_RESULT s2n_early_data_config_free(struct s2n_early_data_config *config)
+{
+    if (config == NULL) {
+        return S2N_RESULT_OK;
+    }
+    GUARD_AS_RESULT(s2n_free(&config->application_protocol));
+    GUARD_AS_RESULT(s2n_free(&config->context));
+    return S2N_RESULT_OK;
+}
+
+int s2n_psk_configure_early_data(struct s2n_psk *psk, uint32_t max_early_data,
+        uint8_t cipher_suite_first_byte, uint8_t cipher_suite_second_byte)
```

Comment:
Is there really no other way to input two bytes of data for the cipher suites? lmao this is so funny.

Reply:
Haha take a look at this: https://godbolt.org/z/fjoPGa

The "just use two uint8s" is the method Cameron and I came up with for https://github.com/awslabs/s2n/commit/016b7d1b4d6464f9fe45dc163ead3e45d610ffe7

We also considered: 
* use a uint16_t: but the IANAS are registered as arrays of two bytes, so it might be unclear which byte of the uint16_t is which byte of the IANA value.
* use an array: but then we have to rely on a customer passing us an array of the right size to fill out

Reply:
works in rust :smile: : https://godbolt.org/z/odPGTz